### PR TITLE
rename edge to mapping

### PR DESCRIPTION
**Pull request by twhetzel**
Added updates based on: https://github.com/INCATools/ontology-access-kit/issues/735#issuecomment-2118298585

There are helper methods that still refer to `Edge`, e.g. "_generate_relation_changes". I'm not sure if those also need to change.

### CODE REVIEW

**Review by cmungall — APPROVED** [review]

File: src/oaklib/interfaces/differ_interface.py [hunk 1]
```diff
@@ -11,9 +11,11 @@
     Change,
     ClassCreation,
     Edge,
-    EdgeChange,
-    EdgeCreation,
     EdgeDeletion,
+    Mapping,
```
Comment (cmungall) [inline src/oaklib/interfaces/differ_interface.py:15]: remove this line

Comment (cmungall) [reply src/oaklib/interfaces/differ_interface.py]: I did this

File: src/oaklib/interfaces/differ_interface.py [hunk 2]
```diff
@@ -275,34 +276,34 @@ def diff(
...
                         predicate=predicate,
                         object=xref,
                     )
-                    yield edge_created
+                    yield mapping_created
             if mappings_removed_set:
                 for mapping in mappings_removed_set:
                     predicate, xref = mapping
-                    deleted_edge = EdgeDeletion(
+                    deleted_mapping = MappingDeletion(
```
Comment (twhetzel) [inline src/oaklib/interfaces/differ_interface.py:289]: @cmungall should `MappingDeletion` be [RemoveMapping](https://incatools.github.io/kgcl/RemoveMapping/)?

File: src/oaklib/interfaces/differ_interface.py [hunk 3]
```diff
@@ -275,34 +276,34 @@ def diff(
...
                         predicate=predicate,
                         object=xref,
                     )
-                    yield deleted_edge
+                    yield deleted_mapping
             if mapping_changed_set:
                 for changes in mapping_changed_set:
                     object, new_predicate, old_predicate = changes
-                    edge_change = EdgeChange(
+                    mapping_change = MappingChange(
```
Comment (twhetzel) [inline src/oaklib/interfaces/differ_interface.py:299]: Should `MappingChange` be [NewMapping](https://incatools.github.io/kgcl/NewMapping/)?

**Comment by twhetzel** [comment]
@hrshdhgd can you take over sorting out what needs to be fixed here?

**Comment by hrshdhgd** [comment]
> @hrshdhgd can you take over sorting out what needs to be fixed here?

Yes, thank you so much for getting this started. I got this from here.

File: src/oaklib/interfaces/differ_interface.py (hunk 3 above)
Comment (hrshdhgd) [reply src/oaklib/interfaces/differ_interface.py]: [MappingReplacement](https://incatools.github.io/kgcl/MappingReplacement/)

File: src/oaklib/interfaces/differ_interface.py (hunk 2 above)
Comment (hrshdhgd) [reply src/oaklib/interfaces/differ_interface.py]: Thank is correct!

**Comment by hrshdhgd** [comment]
This also fixes #732

**Review by cmungall — APPROVED** [review]

**Pull request merged**